### PR TITLE
add overall coverage with ignore files option

### PR DESCRIPTION
**Pull request by peter-crist**

### CODE REVIEW

File: action.yml [hunk 1]
```diff
@@ -49,6 +49,10 @@ inputs:
   trim:
     description: Trim a prefix in the "Impacted Packages" column of the markdown report.
     required: false
+  
+  ignore-files:
+    description: File patterns to ignore from reported overall coverage.
```
Comment (samc1213) [inline action.yml]: Can we indicate this is glob-style vs regexp or something else?

File: cmd/go-coverage-report/main.go [hunk 2]
```diff
@@ -96,7 +99,7 @@ func run(oldCovPath, newCovPath, changedFilesPath string, opts options) error {
 		return nil
 	}
 
-	report := NewReport(oldCov, newCov, changedFiles)
+	report := NewReport(oldCov, newCov, changedFiles) //add ignore here
```
Comment (samc1213) [inline cmd/go-coverage-report/main.go]: ` //add ignore here` - ?

Do we need to do that?

File: cmd/go-coverage-report/overall.go [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+
+	coveragePercentage := float64(coveredStatements) / float64(totalStatements) * 100
+
+	return coveragePercentage, nil
+}
+
+// ignoreFile checks if a file name matches any of the user-specified ignore patterns
+func ignoreFile(fileName string, ignorePatterns []string) bool {
+	if len(ignorePatterns) > 0 {
+	}
```
Comment (samc1213) [inline cmd/go-coverage-report/overall.go]:
```suggestion
```

File: cmd/go-coverage-report/main.go (hunk 2 above)
Comment (peter-crist) [reply cmd/go-coverage-report/main.go]: Nope, old todo comment, thanks.

**Review by samc1213 — APPROVED** [review]

**Pull request merged**